cypress/integration/post_header/post_header_spec.js

```diff
@@ -158,4 +159,58 @@ describe('Post Header', () => {
         // * Check that the RHS is close
         cy.get('#rhsContainer').should('not.be.visible');
     });
+
+    it('should add then remove pinned badge in both center channel and RHS flagged posts list', () => {
+        // 2. Go to a known team and channel
+        cy.visit('/ad-1/channels/town-square');
+
+        // 3. Post a message
+        cy.postMessage('test for pinning/unpinning a post{enter}');
+
+        cy.getLastPostId().then((postId) => {
```

Comment:
As mentioned in DM, please use `getLastPostIdWithRetry` so that it will wait until the last post ID is not on pending state.
```suggestion
        cy.getLastPostIdWithRetry().then((postId) => {
```

Reply:
I don't think this command is already in commands.js @saturninoabril

Reply:
It's [there](https://github.com/mattermost/mattermost-webapp/blob/master/cypress/support/ui_commands.js#L232); please rebase to master.